### ISSUE TITLE
Failed to load pull requests - not a github repo

### ISSUE THREAD

**Issue by nolith** [gitbutlerapp/gitbutler#4191]
### Version

0.12.6

### Operating System

Mac OS X

### Distribution Method

dmg (Apple Silicon)

### Describe the issue

When I open GB it gives the "Failed to load pull requests" error

The screenshot from the developers tools shows that GB is having issue identifying the URL for my project
![image](https://github.com/gitbutlerapp/gitbutler/assets/78752/fc0b2cc2-f989-42a0-8de4-ea28a8679d7d)

I was debugging it and noticed that `.git/gitbutler/virtual_branches.toml` includes the following lines

```
[default_target]
branchName = "main"
remoteName = "origin"
remoteUrl = "file:///Users/nolith/.local/share/my_repos/ringctl.V2YoRlBxej.git"
sha = "3fed3df7a2c35157e963b9cd05169a5ba02ab0f4"
```

As we can see the `remoteUrl` is a file path and GB thinks it is the project name on github.

Moreover, this is no longer the `remoteUrl` or `origin` since a long time, but I have no idea how to refresh it after importing the project.

**Comment by nolith** [comment]
I enabled the experimental feature to change the trunk branch and it refreshed the `remoteUrl`, now I noticed that we are making an assumption that all projects are github projects, cause I see the same error and the development tools shows me the api call made on github for a GitLab hosted project.

**Comment by Byron** [comment]
Thanks so much for digging into this!

There are plenty of things to work on!

* properly classifying URLs as GitHub URLs, which is the only integration that is currently present
* update remote URLs once these change automatically (or maybe don't store them in the first place)

**Comment by krlvi** [comment]
@mtsgrd i think this is related to our conversation about improving the way the github integration is structured and triggered.

**Comment by krlvi** [comment]
@mtsgrd is this one of the issues you fixed before we cut the 0.12.7 release? Perhaps its worth checking if this is still happening

**Comment by nolith** [comment]
> @mtsgrd is this one of the issues you fixed before we cut the 0.12.7 release? Perhaps its worth checking if this is still happening

It's in the changelog and it seems to be working now.

**Comment by krlvi** [comment]
Thanks for verifying this @nolith! In this case, I will close this issue. If you come across any other bugs/issues, opening new gh issues is super appreciated!
